olive/model/__init__.py

```diff
@@ -258,10 +258,8 @@ def __init__(
         self.graph = None
         self.all_graphs: Optional[List[GraphProto]] = None
 
-        # if model_path is local folder, check for onnx file name
-        model_resource_path = self.resource_paths["model_path"]
-        if model_resource_path and model_resource_path.type == ResourceType.LocalFolder:
-            self.get_onnx_file_path(model_resource_path.get_path(), self.onnx_file_name)
+        # check for onnx file name since it will do validation
+        _ = self.model_path
```

Comment:
Will the assignment check the path name aotumatically? A bit confused here.

Reply:
I see, it is a method with property decorator.

Reply:
yes, you can see get_onnx_file_path implementation. I don't change the implementation and just refactoring

Reply:
Yes, you can see the definition of the `model_path` property at line 302

Reply:
I suppose @jambayk add this method?

Reply:
Yes thanks, I get the checking callback stack now.